src/Microsoft.OData.Core/UriParser/SemanticAst/SelectExpandClauseExtensions.cs

```diff
@@ -83,7 +83,20 @@ internal static void GetSelectExpandPaths(this SelectExpandClause selectExpandCl
         /// <returns>String list generated from selected items</returns>
         internal static List<string> GetCurrentLevelSelectList(this SelectExpandClause selectExpandClause)
         {
-            return selectExpandClause.SelectedItems.Select(GetSelectString).Where(i => i != null).ToList();
+            if (selectExpandClause.AllAutoSelected)
+            {
+                return selectExpandClause.SelectedItems
+                    .OfType<ExpandedNavigationSelectItem>()
+                    .Select(i => String.Join("/", i.PathToNavigationProperty.WalkWith(PathSegmentToStringTranslator.Instance).ToArray()))
+                    .ToList();
+                //return new List<string>();
+            }
+            else
+            {
+                return selectExpandClause.SelectedItems
+                    .Select(GetSelectString)
+                    .Where(i => i != null).ToList();

```

Comment:
I would recommend breaking the line before dot to align `ToList` directly below `Where` to form nice chain.